server/db/mongodb/adapter.go

```diff
@@ -0,0 +1,1521 @@
+// +build mongodb
+
+package mongodb
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"hash/fnv"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/tinode/chat/server/auth"
+	"github.com/tinode/chat/server/store"
+	t "github.com/tinode/chat/server/store/types"
+	b "go.mongodb.org/mongo-driver/bson"
+	mdb "go.mongodb.org/mongo-driver/mongo"
+	mdbopts "go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// adapter holds MongoDB connection data.
+type adapter struct {
+	conn       *mdb.Client
+	db         *mdb.Database
+	dbName     string
+	maxResults int
+	version    int
+}
+
+const (
+	defaultHost     = "localhost:27017"
+	defaultDatabase = "tinode"
+
+	adpVersion  = 108
+	adapterName = "mongodb"
+
+	defaultMaxResults = 1024
+)
+
+var ctx context.Context
+
+// See https://godoc.org/go.mongodb.org/mongo-driver/mongo/options#ClientOptions for explanations.
+type configType struct {
+	Addresses      interface{} `json:"addresses,omitempty"`
+	ConnectTimeout int         `json:"timeout,omitempty"`
+
+	// Options separately from ClientOptions (custom options):
+	Database string `json:"database,omitempty"`
+}
+
+// Open initializes mongodb session
+func (a *adapter) Open(jsonconfig json.RawMessage) error {
+	if a.conn != nil {
+		return errors.New("adapter mongodb is already connected")
+	}
+
+	var err error
+	var config configType
+	if err = json.Unmarshal(jsonconfig, &config); err != nil {
+		return errors.New("adapter mongodb failed to parse config: " + err.Error())
+	}
+
+	var opts mdbopts.ClientOptions
+
+	if config.Addresses == nil {
```

Comment:
You probably want to add login/password as configurable options.

Reply:
Please add a .md file describing your storage schema, like https://github.com/tinode/chat/blob/master/server/db/rethinkdb/schema.md or https://github.com/tinode/chat/blob/master/server/db/mysql/schema.sql